src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -627,6 +629,219 @@ class Polyhedron
     return has_nbrs;
   }
 
+  /*!
+ * \brief Creates a Polyhedron from a given Hexahedron's vertices.
+ *
+ * \param [in] hex The hexahedron
+ * \param [in] checkSign If true (default is false), checks the volume of the
+ *             Polyhedron is positive. If volume is negative, order of some
+ *             vertices will be swapped.
+ *
+ * \return A Polyhedron with the Hexahedron's vertices and added
+ *         vertex neighbors
+ *
+ * \note The Hexahedron is assumed to have a specific vertex order:
+ * \verbatim
+ *
+ *          7--------6          +y
+ *         /|       /|               +z
+ *        / |      / |           ^  >
+ *       3--------2  |           | /
+ *       |  4-----|--5           |/
+ *       | /      | /            -----> +x
+ *       |/       |/
+ *       0--------1
+ *
+ * \endverbatim
+ *
+ *       The Polyhedron's vertex neighbors are created assuming this vertex
+ *       ordering.
+ *
+ * \note checkSign flag does not guarantee the Polyhedron's vertex order
+ *       will be valid. It is the responsiblity of the caller to pass
+ *       a Hexahedron with a valid vertex order.
+ */
+  AXOM_HOST_DEVICE
+  static Polyhedron from_primitive(const Hexahedron<T, NDIMS>& hex,
```

Comment:
Thanks for adding these `from_primitive` helper functions! It should make it a lot easier for users to setup these common shapes.